keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +464,136 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
+
 #### Container Lifecycle 
 - The container's lifecycle will be managed by the container runtime just like other Windows container types.
-#### Container users 
--  The privileged container should be able to run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+
+#### Container users
+
+- The privileged container can run as any user that's available on the host or in the domain of the host machine. Password accounts are being investigated.
+Running privileged containers as non SYSTEM/admin accounts will be the primary way operators can restrict access to system resources (files, registry, named pipes, WMI, etc).
+More information on Windows resource access can be found at https://docs.microsoft.com/en-us/archive/msdn-magazine/2008/november/access-control-understanding-windows-file-and-registry-permissions.
+
 #### Container Mounts
-- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.  
+- Directory mounts (i.e. directory C:\test mapped to C:\test in the privileged container) are still being investigated. Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container. We are still investigation how external mounts such as secrets or storage accounts would be added. The default visibility of all the host filesystem may present backwards compatibility challenges in the future if host path mounting is enabled in Windows. Currently the only method to restrict would be via the user in a way similar to [UID or GID](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) in the PSP. Currently however, this host path mounting is unlikely to be enabled in the future as a restriction of Job Objects.
```

Comment:
>  Note that file system mapping in the current implementation would expose both the host filesystem and the container filesystem to the processes in the privileged container.

Can you elaborate more about this? Privileged container will have the full root file system mounted into the container?

Reply:
So if you look at the [design overview](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#overview) in this KEP that was merged previously in a proposed state there are some details here.
These "privileged" containers are actually just windows OS job objects that run will run on the host instead of in a server silo so standard containers. Because of this there the container will have access to the entire host/root file system without anything actually being mounted.
To address security concerns here the guidance will be to run privileged containers as specific users and ACL host resources accordingly (this guidance is already mentioned in these updates)

Does that help answer your question?

Reply:
In that case, the artifacts in the container image will be installed directly to locations in the host/root file system too?

Just for my education. :)

Reply:
Artifacts in the container will show up in a new volume mounted at c:\c\{guid} (this is either containerid or job object id, I forget which one) on the host. 
These volumes get deleted with the containers.

Reply:
I see. Will there be a "chroot"? Within the container does it see the host root as its root? Or does it have its own root?

If it doesn't have its own root, how does its own files? e.g. do they need to know the container ID?
If it has its own root, how does it go out and see the host root, given that you mentioned that `will have access to the entire host/root file system`.

Reply:
I'm not sure how this maps to roots because I think this all works differently on Windows.

The containers have a default working directory that is set to the container volume (`c:\\c\\{id}`)
%SYSTEMDRIVE% and %SYSTEMROOT% get set to `c:` and `c:\windows` respectively.

If a container is bringing in files they can be accessed with a relative path to those files.
If the container wants to access host files it can use an absolute path to those files.
Standard windows PATH searching will be based on host paths by default (because by default PATH search folders are absolute).

Does this help?

Reply:
I see. So the current working directory will be switched to the container volume. Thanks for the explanation!